src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala

```diff
@@ -122,4 +153,48 @@ trait OAuthService extends HttpService with PerRequestCreator with FireCloudDire
       }
     }
 
+
+  private def initiateAuth(state: String, approvalPrompt: String) =  {
+    log.info(s"initiateAuth: $approvalPrompt")
+    val gcsAuthUrl = HttpGoogleServicesDAO.getGoogleRedirectURI(state, approvalPrompt=approvalPrompt)
+    redirect(gcsAuthUrl, StatusCodes.TemporaryRedirect)
+  }
+
+  private def completeAuthWithRedirect(actualState: String, accessToken: String) = {
+    // as created in the /login endpoint, the actual state contains the desired callback url
+    // and the desired callback fragment, separated by a hash. But, either value may be the empty string.
+
+    val redirectParts = actualState.split("#")
```

Comment:
... Replace `"{access_token}"` with `accessToken` in `actualState`, then parse it as a URL.
Check the parts you want against the whitelist, redirect.